libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1452,4 +1467,89 @@ public static String markMediaFailed(Context context, @NonNull String postConten
         }
         return postContent;
     }
+
+    public static boolean hasMediaItemsMarkedUploading(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_UPLOADING);
+    }
+
+    public static boolean hasMediaItemsMarkedFailed(Context context, @NonNull String postContent) {
+        return hasMediaItemsMarkedWithTag(context, postContent, ATTR_STATUS_FAILED);
+    }
+
+    private static boolean hasMediaItemsMarkedWithTag(Context context, @NonNull String postContent, String tag) {
+        // fill in Aztec with the post's content
+        AztecText content = new AztecText(context);
```

Comment:
Wonder why if it's easier to just get all the media items in the editor and check for the presence of attribute. Without using Predicates.

```AztecMediaSpan[] imageOrVideoSpans = content.getText().getSpans(0, content.getText().length(), AztecMediaSpan.class);
                    for (AztecMediaSpan currentClass: imageOrVideoSpans) {
                      // do the check here
                    }
```